pkg/cloudprovider/provider/gce/service.go

```diff
@@ -40,15 +44,23 @@ func connectComputeService(cfg *config) (*service, error) {
 
 // networkInterfaces returns the configured network interfaces for an instance creation.
 func (svc *service) networkInterfaces(cfg *config) ([]*compute.NetworkInterface, error) {
+	network := cfg.network
+
+	if cfg.network == "" && cfg.subnetwork == "" {
+		network = defaultNetwork
+	}
+
 	ifc := &compute.NetworkInterface{
 		AccessConfigs: []*compute.AccessConfig{
 			{
 				Name: "External NAT",
 				Type: "ONE_TO_ONE_NAT",
 			},
 		},
-		Network: "global/networks/default",
+		Network:    network,
```

Comment:
This means the user needs to enter the 'full path' to the actual network?
Like the default network is `global/networks/default`

Reply:
ether of "name" or "self_links" are documented as working solution.

Reply:
Oh great, thanks!

Reply:
full self_links will be fed from terraform as it has only `.self_link` property as a way to address networks/subnetworks, but simple names will work too.